packages/ui/src/elements/Upload/index.tsx

```diff
@@ -296,9 +330,33 @@ export const Upload_v4: React.FC<UploadProps_v4> = (props) => {
 
   const imageCacheTag = uploadConfig?.cacheTags && savedDocumentData?.updatedAt
 
-  if (uploadConfig.hideFileInputOnCreate && !savedDocumentData?.filename) {
-    return null
-  }
+  useEffect(() => {
+    void (async () => {
+      const slots = await getDocumentSlots({ collectionSlug })
+      setDocumentSlots(slots)
+    })()
+  }, [getDocumentSlots, collectionSlug])
```

Comment:
You should be able to get these from props. The DefaultEditView is provided `UploadControls` as a prop. So you should thread that through to the upload component to avoid fetching here in an effect.

Reply:
- Moved the effect to DefaultEditView where the other effects are defined.
- Added `documentSlots` as a prop on Upload.